tests/integration/test_integration.py

```diff
@@ -72,102 +72,100 @@ async def initialize_vault(vault: Vault, timeout: int = 60) -> Tuple[str, str]:
         Retrieves Vault's LoadBalancer address, initializes Vault and generates a token for
         the charm.
 
         Args:
-            namespace (str): Kubernetes namespace
+            ops_test: Ops test Framework.
 
         Returns:
             str: Generated token.
         """
-        kubernetes = Kubernetes(namespace=namespace)
+        kubernetes = Kubernetes(namespace=ops_test.model_name)  # type: ignore[arg-type]
         load_balancer_address = await self.wait_for_load_balancer_address(kubernetes=kubernetes)
         vault = Vault(url=f"http://{load_balancer_address}:8200")
-        unseal_key, root_token = await self.initialize_vault(vault=vault)
+        unseal_key, root_token = await self.initialize_vault(vault=vault)  # type: ignore[misc]
         vault.set_token(root_token)
         vault.unseal(unseal_key=unseal_key)
         generated_token = vault.generate_token(ttl="5m")
         return generated_token
 
-    async def test_given_no_config_when_deploy_then_status_is_waiting(  # noqa: E501
-        self, ops_test, charm, cleanup
+    @pytest.mark.abort_on_fail
+    async def test_given_no_config_when_deploy_then_status_is_blocked(  # noqa: E501
+        self, ops_test: OpsTest, build_and_deploy
     ):
-        await self.deploy_charm(ops_test, charm)
-
-        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="blocked", timeout=1000)
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="blocked", timeout=1000)  # type: ignore[union-attr]
 
+    @pytest.mark.abort_on_fail
     async def test_given_no_config_when_post_deployment_tasks_and_authorise_charm_then_status_is_active(  # noqa: E501
-        self, ops_test, charm, cleanup
+        self, ops_test: OpsTest, build_and_deploy
     ):
-        """This test follows the README.MD deployment and post-deployment tasks.
+        """This test follows the README.md deployment and post-deployment tasks.
 
         Args:
-            ops_test: Ops test Framework
-            charm: Charm path
-
-        Returns:
-            None
+            ops_test: Ops test Framework.
+            build_and_deploy: Pytest fixture.
+            post_deployment_tasks: Pytest fixture.
         """
-        await self.deploy_charm(ops_test, charm)
-        vault_unit = ops_test.model.units["vault-k8s/0"]
-        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="blocked", timeout=1000)
-
-        vault_token = await self.post_deployment_tasks(namespace=ops_test.model_name)
+        vault_unit = ops_test.model.units["vault-k8s/0"]  # type: ignore[union-attr]
 
+        vault_token = await self.post_deployment_tasks(ops_test)
         await vault_unit.run_action(action_name="authorise-charm", token=vault_token)
-        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1000)
 
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1000)  # type: ignore[union-attr]
```

Comment:
Isn't lint going crazy about the line being too long?

Reply:
No, but I fixed it and it's not complaining.